Civi/Contract/ActionProvider/Action/GetContract.php

```diff
@@ -166,6 +167,8 @@ function str_starts_with($str, $start) {
       $output->setParameter('frequency_unit', $recurring_contribution['frequency_unit']);
       $output->setParameter('frequency_interval', $recurring_contribution['frequency_interval']);
 
+      $bankaccount_reference = \civicrm_api3('BankingAccountReference', 'getSingle', ['ba_id' => $contract[U::getCustomFieldId('membership_payment.from_ba')], 'reference_type_id' => U::lookupOptionValue('civicrm_banking.reference_types', 'iban', 'id'), 'return' => ["reference"]]);
```

Comment:
I'm not sure that you can just assume that this is definitely an IBAN, even though we mostly use it that way. I would either make sure that it *is* an IBAN, or refer to it as a bank account reference.

Reply:
I tried to make sure it is an iban by setting reference_type_id to the option-value of IBAN.
Is this not sufficient? What else could I do?

Reply:
But another problem might be, that getSingle returns an error message if there is no or there are several BankingAccountReferences available for the type IBAN.

Reply:
ok. I changed it to from getSingle to get-request